### PR TITLE
Add jenkins 2 article

### PR DESCRIPTION
**Pull request by gianarb**

### CODE REVIEW

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+img: /img/docker.png
+tags: docker, oss, jenkins
+summary: "A little overview about Jenkins 2 but the main topic of the article
+is about how run docker inside docker to start a continuous integration system
+inside a container"
+priority: 0.6
+changefreq: yearly
+---
+
+Docker inside docker and Jenkins 2 Jenkins  is one of the most famous
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: I think, the Sentence is starting at `Jenkins is` either have a `.`  after 2 or remove it

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+continuous integration and deployment tools, it’s written in Java and it helps
+you to manage your pipeline and all tasks that help you to put your code in
+production or manage your build.
+
+I am happy to announce that the version 2, in
+my opinion is one of the best discoveries in the last year.
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: version 2 is a discovery of this year, right?

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+priority: 0.6
+changefreq: yearly
+---
+
+Docker inside docker and Jenkins 2 Jenkins  is one of the most famous
+continuous integration and deployment tools, it’s written in Java and it helps
+you to manage your pipeline and all tasks that help you to put your code in
+production or manage your build.
+
+I am happy to announce that the version 2, in
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: The announcement of Jenkins release of `version 2` few days ago, is one of the best release of this year in my opinion.

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+my opinion is one of the best discoveries in the last year.
+
+* Nice installation wizard Refactoring of the design, one of the most critical
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Though, we all compromised with few features missing in the version 1 of Jenkins to get the best of this powerful tool. I am happy that the `version 2` has added those features into the list. Like,

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+---
+
+Docker inside docker and Jenkins 2 Jenkins  is one of the most famous
+continuous integration and deployment tools, it’s written in Java and it helps
+you to manage your pipeline and all tasks that help you to put your code in
+production or manage your build.
+
+I am happy to announce that the version 2, in
+my opinion is one of the best discoveries in the last year.
+
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Though, we all compromised with few features missing in the version 1 of Jenkins to get the best of this powerful tool. I am happy that the version 2 has added those features into the list. Like,
 Add a line note

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown (hunk 4 above)
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Nice installation wizard `,` ...

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+* Nice installation wizard Refactoring of the design, one of the most critical
+  feature of the previous version
+* Good and modern set of plugins like [Jenkins
+  Pipeline](https://jenkins.io/solutions/pipeline/) to manage your build
+
+Jenkins is a switty revelation but the tool of the moment it’s docker, engine
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Not sure what switty revelation means! :(

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+that allow you to work easier with the containers.
+
+This two tools together are perfect to create your isolated environment to test
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: to create an isolated environment.

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+and deploy your applications.
+
+The first setup could be install Jenkins on your
+server and use a plugin to manage the integration and trigger your test inside
+an isolated environment, the container.
+
+Great work but in my opinion reproducibility is one of the critical point when
+you deal with plugins if you can not run your build on your local environment
+easy you have a problem.  Secondly if the container could be a good solution to
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: easily? then you have a problem

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+deploy and maintain a solid and isolated application, why your Jenkins has not
+the privilege to run inside a container?  In this perspective how can we can
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: In this perspective, how can we .....

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+run container inside a container?
+
+Ok it’s our time, we have a list of problems to solve.
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Ok, now its the time to figure it out how to solve the problems...

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+We can use the official Jenkins image to put jenkins inside a container, but I
+worked on my personal alpine installation, light and easy, [this is the
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Here, is the dockerFile

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    gianarb/jenkins:2.0
+{% endhighlight %}
+
+We connect on `http://docker-ip:8080` and start the new awesome wizard!
+
+<img class="img-responsive" alt="First Jenkins 2 page, grab from the log your key and start" src="/img/docker-in-docker/jenkins2-start.png">
+
+<img class="img-responsive" alt="Jenkins's plugins wizard" src="/img/docker-in-docker/jenkins2-plugin.png">
+
+To verify that all work we can create a new job it onlu runs `docker ps -a` our
```
Comment (suresh1209) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: typo `only`

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown (hunk 1 above)
Comment (gianarb) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: I drunk a lot!

File: _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown (hunk 2 above)
Comment (gianarb) [inline _posts/2016-04-10-docker-inside-docker-and-jenkins-2.markdown]: Yes :)

**Pull request merged**